src/DataCollection.Shared/ViewModels/DestinationRelationshipViewModel.cs

```diff
@@ -97,44 +96,92 @@ public async Task LoadViewModel(RelatedFeatureQueryResult relatedFeatureQueryRes
         /// <summary>
         /// Gets or sets the collection of available values to select from for the related record
         /// </summary>
-        public IOrderedEnumerable<PopupManager> OrderedAvailableValues { get; private set; }
+        public IEnumerable<PopupManager> OrderedAvailableValues
+        {
+            get => _availableValues;
+            private set
+            {
+                _availableValues = value;
+                OnPropertyChanged();
+            }
+        }
 
         /// <summary>
-        /// Get all of the values in the related table to display when editing feature
+        /// Tracks whether a refresh of <see cref="OrderedAvailableValues"/> is in progress
         /// </summary>
-        private async Task GetAvailableValues()
+        public bool IsRefreshingValues
         {
-            var queryParams = new QueryParameters()
+            get => _isRefreshingValues;
+            set
             {
-                ReturnGeometry = false,
-                WhereClause = "1=1",
-            };
+                if (_isRefreshingValues != value)
+                {
+                    _isRefreshingValues = value;
+                    OnPropertyChanged();
+                    (RefreshValuesCommand as DelegateCommand)?.RaiseCanExecuteChanged();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Starts a refresh of the <see cref="OrderedAvailableValues"/> collection.
+        /// </summary>
+        public ICommand RefreshValuesCommand => _refreshValuesCommand ?? (_refreshValuesCommand = new DelegateCommand(parm =>
+		    {
+	            _ = RefreshAvailableValues(true).ContinueWith(res => LoadSelectedFeaturePopup());
+		    },
+            (param) =>
+            {
+                return !IsRefreshingValues;
+            })
+        );
+
+        /// <summary>
+        /// Loads <see cref="OrderedAvailableValues"/> with popups for all of the entries in the relevant table.
+        /// Values loaded from cache if <paramref name="invalidateCache"/> is false, otherwise cache is overwritten.
+        /// </summary>
+        private async Task RefreshAvailableValues(bool invalidateCache = false)
+        {
+            // Ensures that cache is only updated/checked by one Task at a time
+            await cacheMutex.WaitAsync();
 
             try
             {
+                if (!invalidateCache && CachedTableResults.ContainsKey(FeatureTable))
+                {
+                    OrderedAvailableValues = CachedTableResults[FeatureTable];
```

Comment:
To avoid raising a PropertyChangedEvent every time RefreshAvailableValues(false) is called, don't call the setter of OrderedAvailableValues when it already has a value:
```suggestion
                    if (OrderedAvailableValues == null)
                        OrderedAvailableValues = CachedTableResults[FeatureTable];
```